Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,869 @@ public override bool IsServerConnected()
+            applicationModule.Name = appModuleNameTobeCreated;
+            applicationModule.SetValue("description", desc);
+
+            applicationModule.SetValue("parent", new BaseEntity("application_module")
+            {
+                Id = paraentId,
+                TypeName = "application_module"
+            });
+
+            ApplicationModule module = Task.Run(() =>
+            {
+                return this.octaneRepository.CreateEntity<ApplicationModule>(GetLoginDTO(), applicationModule, new List<string>() { "path", "id" });
+            }).Result;
+
+            return module.Id.ToString();
+        }
+
+        public bool ExportActivitiesGroupToALM(ActivitiesGroup activitiesGroup, QCTestCase mappedTest, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, ObservableList<ExternalItemFieldBase> designStepsFields, ObservableList<ExternalItemFieldBase> designStepsParamsFields, ref string result)
+        {
+            try
+            {
+                string step = string.Empty;
+                activitiesGroup.ActivitiesIdentifiers.ToList().ForEach(p =>
+                {
+                    if (!string.IsNullOrEmpty(step))
+                    {
+                        step += "- ";
+                    }
+                    step += p.ActivityName + " ";
+
+                    if (p.IdentifiedActivity.Variables.Any())
+                    {
+                        p.IdentifiedActivity.Variables.ToList().ForEach(f =>
+                        {
+                            step += " <" + f.Name + "> ";
+                        });
+                    }
+                    step += '\n';
+                });
+
+                if (mappedTest == null) //#Create new test case
+                {
+                    CreateNewTestCase(activitiesGroup, fatherId, testCaseFields, step);
+                }
+                else //##update existing test case
+                {
+                    //TODO: Maheshk: Update existing testcase
+                    if (!string.IsNullOrEmpty(activitiesGroup.ExternalID))
+                    {
+                        UpdateTestCase(activitiesGroup, fatherId, testCaseFields, step);
+                    }
+                    else
+                    {
+                        CreateNewTestCase(activitiesGroup, fatherId, testCaseFields, step);
+                    }
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the " + GingerDicser.GetTermResValue(eTermResKey.ActivitiesGroup) + " to QC/ALM", ex);
+                return false;
+            }
+        }
+
+        private string CreateNewTestCase(ActivitiesGroup activitiesGroup, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, string testScript)
```

Comment:
Codacy found an issue: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)